src/contracts/Gateway/AbstractDoctrineDatabase.php

```diff
@@ -125,18 +125,15 @@ public function countAll(): int
     public function countBy($criteria): int
     {
         $metadata = $this->getMetadata();
-        $qb = $this->connection->createQueryBuilder();
+        $qb = $this->createBaseQueryBuilder();
         $this->applyInheritance($qb);
 
         $identifierColumn = $metadata->getIdentifierColumn();
         $tableAlias = $this->getTableAlias();
-        $qb->select($this->connection->getDatabasePlatform()->getCountExpression($tableAlias . '.' . $identifierColumn));
-        $qb->from($metadata->getTableName(), $tableAlias);
+        $platform = $this->connection->getDatabasePlatform();
+        $qb->select($platform->getCountExpression(sprintf('DISTINCT %s.%s', $tableAlias, $identifierColumn)));
```

Comment:
There's `\Doctrine\DBAL\Query\QueryBuilder::distinct` but maybe not usable here given it's `COUNT(DISTINCT )` not `DISTINCT COUNT()`?

Reply:
Yes, due to the above it is not usable. I have actually checked and we have a similar code in our `ibexa/core`, and this syntax works for our supported databases.